.github/workflows/debian-package.yml

```diff
@@ -0,0 +1,77 @@
+name: Debian package build
+
+on:
+  push:
+#    branches: [ main ]
+    paths-ignore:
+      - '.clang-format'
+      - '.mdl-styles'
+      - '*.md'
+      - 'docs/**'
+      - 'licenses/**'
+      - 'website/**'
+
+  pull_request:
+#    branches: [ main ]
+    paths-ignore:
+      - '.clang-format'
+      - '.mdl-styles'
+      - '*.md'
+      - 'docs/**'
+      - 'licenses/**'
+      - 'website/**'
```

Comment:
Thinking we probably don't need to generate .deb files for every branch, and instead only produce debs from `main` branch commits.

if the team agrees I'll comment-in these two branch lines in before merge.

Reply:
Oh, main is definitely more than enough 😄 Maybe only tagged releases would be enough, so not on every commit?

Reply:
Yes; maybe not every commit 😄 , but this would build a deb at least for the PR merges to main, so more like "every feature". 
(There's a lot of active debian PPA packages that track upstream maik branches, so this feel about right and matches how the website let's people download zips of the main)

Reply:
If it runs quickly and it's cheap, why not, yeah. If it's long-running and heavy, we'll need to discuss this further. After all, it's a relatively niche thing, so we don't want it to cause problems and slowdowns.

Reply:
> If it runs quickly and it's cheap, why not, yeah.

(:+1: - it completes faster than all the workflows; it's only a single build)

Updated to only run after PR's are merged to `main`, so basically the Debian-equivalent for how Mac users getting a .dmg and Windows users getting a setup.exe that tracks the `main` branch.

Debian-based distros are (I think) the most popular, so this hopefully will be a good option for those who don't have a build environment.

Reply:
> Updated to only run after PR's are merged to main, so basically the Debian-equivalent for how Mac users getting a .dmg and Windows users getting a setup.exe that tracks the main branch.

Yeah, seem reasonable.

> Debian-based distros are (I think) the most popular, so this hopefully will be a good option for those who don't have a build environment.

Cool, just pointing out we've been publishing Linux dev build binaries for quite a while now:
https://github.com/dosbox-staging/dosbox-staging/actions/runs/9771784818

So grabbing simply the binaries has always been an option regardless of what Linux distro you're on, and continues to be an option.

Reply:
> So grabbing simply the binaries has always been an option regardless of what Linux distro 

Understood; zip and deb have the same binary inside.

(The only benefit here is just a bit more integration for Debian-based users:  their package manager will download and install all the dependencies and `dosbox` is added to their path; these are minor but some people like those freebies :laughing: )